tex2wiki/src/tex2wiki.py

```diff
@@ -0,0 +1,618 @@
+            elif dollar_count > 1:
+                result = result[:-1] + "<br />\n<math>{\\displaystyle "
+            else:
+                result += "<math>{\\displaystyle "
+            dollar_count += 1
+        else:
+            result += ch
+
+    return result.strip()
+
+
+def extract_data(data):
+    # type: (list) -> list
+    """Extracts the equations and pertinent data from the tex file."""
+    result = list()
+
+    for section_data in data:
+        equations = section_data[1][:-1]
+        for i, equation in enumerate(equations):
+            # formula stuff
+            try:
+                formula = format_formula(get_data_str(equation, latex="\\formula"))
+            except IndexError:  # there is no formula.
+                break
+
+            equation = equation.split("\n")
+
+            # get metadata
+            raw_metadata = ""
+
+            j = 0
+            while j < len(equation):
+                if "%" in equation[j]:
+                    raw_metadata += equation.pop(j)[1:].strip().strip("\n") + "\n"
+                else:
+                    j += 1
+
+            metadata = dict()
+            for data_type in METADATA_TYPES:
+                metadata[data_type] = format_metadata(get_data_str(raw_metadata, latex="\\"+data_type))
+
+            equations[i] = LatexEquation(formula, '\n'.join(equation), metadata)
+
+        result.append([section_data[0], equations])
+
+    return result
+
+
+def find_end(text, left_delimiter, right_delimiter, start=0):
+    # type: (str, str, str(, int)) -> int
+    """A .find that accounts for nested delimiters."""
+    net = 0  # left delimiters encountered - right delimiters encountered
+    for i, ch in enumerate(text[start:]):
+        if ch == left_delimiter:
+            net += 1
+        elif ch == right_delimiter:
+            if net == 1:
+                return i + start
+            else:
+                net -= 1
+
+    return -1
+
+
+def get_data_str(text, latex=""):
+    # type: (str(, str)) -> str
+    """Gets the string in between curly brackets."""
+    start = text.find(latex + "{")
+
+    if start == -1:
+        return ""
+
+    return text[start + len(latex + "{"):find_end(text, "{", "}", start)]
+
+
+def generate_nav_bar(info):
+    # type: (list) -> list
+    """Generates the navigation bar code for a page."""
+    links = list()
+    for link, text in info:
+        link = link.replace("''", "")
+        text = text.replace("''", "")
+        links.append(generate_link(link, text))
+
+    nav_section = generate_html("div", "<< " + links[0], options={"id": "alignleft"}, spacing=1) + "\n" + \
+        generate_html("div", links[1], options={"id": "aligncenter"}, spacing=1) + "\n" + \
+        generate_html("div", links[2] + " >>", options={"id": "alignright"}, spacing=1)
+
+    header = generate_html("div", nav_section, options={"id": "drmf_head"})
+    footer = generate_html("div", nav_section, options={"id": "drmf_foot"})
```

Comment:
I get the impression that this does not generate a template based footer as done here https://github.com/DRMF/DRMF-Seeding-Project/pull/76/files#diff-38aefdb6ba2f5d900522ba955ff33793L414